generators/typescript/utils/core-utilities/fetcher/src/form-data-utils/FormDataWrapper.ts

```diff
@@ -1,72 +1,144 @@
-import { Readable } from "stream";
 import { RUNTIME } from "../runtime";
 
-interface CrossPlatformFormData {
-    append(key: string, value: any): void;
+export type MaybePromise<T> = Promise<T> | T;
+
+type FormDataRequest<Body> = {
+    body: Body;
+    headers: Record<string, string>;
+};
+
+export interface CrossPlatformFormData {
+    setup(): Promise<void>;
+
+    append(key: string, value: unknown): void;
+
+    appendFile(key: string, value: unknown, fileName?: string): Promise<void>;
+
+    getRequest(): MaybePromise<FormDataRequest<unknown>>;
+}
+
+export async function newFormData(): Promise<CrossPlatformFormData> {
+    let formdata: CrossPlatformFormData;
+    if (RUNTIME.type === "node" && RUNTIME.parsedVersion != null && RUNTIME.parsedVersion > 18) {
+        formdata = new Node19FormData();
+    } else if (RUNTIME.type === "node") {
+        formdata = new Node16FormData();
+    } else {
+        formdata = new WebFormData();
+    }
+    await formdata.setup();
+    return formdata;
 }
 
-class FormDataRequestBody {
-    private fd: any;
-    private encoder: any;
+/**
+ * Form Data Implementation for Node.js 18+
+ */
+class Node19FormData implements CrossPlatformFormData {
+    getBody(): unknown {
+        throw new Error("Method not implemented.");
+    }
+    getHeaders(): MaybePromise<Record<string, string>> {
+        throw new Error("Method not implemented.");
+    }
+    private fd:
+        | {
+              append(name: string, value: unknown, fileName?: string): void;
+          }
+        | undefined;
 
-    constructor(fd: any) {
-        this.fd = fd;
+    public async setup() {
+        this.fd = new (await import("formdata-node")).FormData();
     }
 
-    async setup(): Promise<void> {
-        if (this.encoder == null && RUNTIME.type === "node") {
-            this.encoder = new (await import("form-data-encoder")).FormDataEncoder(this.fd);
-        }
+    public append(key: string, value: any): void {
+        this.fd?.append(key, value);
+    }
+
+    public async appendFile(key: string, value: any, fileName?: string): Promise<void> {
+        this.fd?.append(key, new (await import("buffer")).Blob([value]), fileName);
     }
 
-    /**
-     * @returns the multipart form data request
-     */
-    public async getBody(): Promise<any> {
-        if (RUNTIME.type !== "node") {
-            return this.fd;
+    public async getRequest(): Promise<FormDataRequest<unknown>> {
+        if (!this.fd) {
+            throw new Error("Form data is not defined");
```

Comment:
Okay to skip for now but we should be pretty careful of throwing an `Error` -- its pretty hard for TypeScript consumers to catch. Our TypeScript SDKs generally have a top level error class (e.g. `CohereError`) that makes it easy for consumers to catch any sort of exception.